akka-persistence-typed/src/main/scala/akka/persistence/typed/EventSourcedSignal.scala

```diff
@@ -29,6 +31,56 @@ final case class RecoveryFailed(failure: Throwable) extends EventSourcedSignal {
   def getFailure(): Throwable = failure
 }
 
+final case class PersistFailed[Command, Event](failure: Throwable, command: Option[Command], event: Event)
+    extends EventSourcedSignal {
+
+  /**
+   * Java API
+   */
+  def getFailure(): Throwable = failure
+
+  /**
+   * Java API
+   */
+  def getCommand(): Optional[Command] = {
```

Comment:
In which case can we have a `None`? When replicating?

Reply:
yes, when replicated event. I added some api docs